src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/RedditController.java

```diff
@@ -2,9 +2,40 @@
 
 import org.springframework.web.bind.annotation.RestController;
 
+import edu.ucsb.cs156.spring.backenddemo.services.RedditQueryService;
+import lombok.extern.slf4j.Slf4j;
 
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+@Api(description="Subreddit info from Reddit")
+@Slf4j
 @RestController
+@RequestMapping("/api/reddit")
 public class RedditController {
-    
+    ObjectMapper mapper = new ObjectMapper();
+
+    @Autowired
+    RedditQueryService redditQueryService;
+
+    @ApiOperation(value = "Get subreddit data from a certain Reddit subreddit", notes = "JSON return format documented here: https://www.reddit.com/r/subreddit.json")
```

Comment:
We might want to make the operation description match the staff solution, and I think the staff solution doesn't have a note

Reply:
@eduglt Fixed